src/main.rs

```diff
@@ -347,6 +348,13 @@ fn main() {
 
 	let mut logger = Logger::new(silence_level, &rom_path);
 
+	// create dump dir if it does not exist already
+	if let Some(ref dump_dir) = cli.dump_dir {
+		if let Err(msg) = fs::create_dir_all(dump_dir) {
```

Comment:
maybe add logic to avoid creating a directory if one already exists, to avoid an error. not mandatory.

Reply:
The `fs::create_dir_all` methods already does that. From its documentation:

    Recursively create a directory and all of its parent components if they are missing.

Reply:
The error it would have returned is only for other stuff, like insufficient permissions. But handling such issues should be left to the user I think.

Reply:
One thing I could maybe do is treat failure to create a dump directory as a fatal error and exit the program. I think it would be quite reasonable to do that, since the user explicitly states that he/she wants a dump directory, but the program was not able to create one. That would be a problem imo. What are your thoughts?

Reply:
imo you are right and can close the issue

Reply:
I've added a call to `exit` then to quit the program on any failure to create the dump dir as discussed. Closing the issue.